### PR TITLE
Calling v3

### PR DESCRIPTION
**Pull request by deanrobertcook**
Please check also all sounds in the app and also calling v2 - there was a fair bit of refactoring done there

To enable v3, please go to developer settings and look for the check box. The outgoing call will always be whatever is set in settings, but the device should be able to accept either a v2 or a v3 call regardless of whether the box is checked or not.

If I screwed up the dependencies, nothing will happen...
#### APK
[Download build #8330](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8330/artifact/build/artifact/wire-dev-PR429-8330.apk)
[Download build #8331](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8331/artifact/build/artifact/wire-dev-PR429-8331.apk)
[Download build #8333](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8333/artifact/build/artifact/wire-dev-PR429-8333.apk)
[Download build #8334](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8334/artifact/build/artifact/wire-dev-PR429-8334.apk)
[Download build #8336](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8336/artifact/build/artifact/wire-dev-PR429-8336.apk)
[Download build #8338](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8338/artifact/build/artifact/wire-dev-PR429-8338.apk)

### CODE REVIEW

**Comment by deanrobertcook** [comment]
build again

**Comment by deanrobertcook** [comment]
build again

**Comment by wmengli** [comment]
Video call - video frame frozen. No incoming video received and seems no outgoing video sent too
Build #8161, Sony Z3, Android 6.0
C1@Android, C2@iOS or Wrapper
1. C1@Android using calling v2. C1 makes outgoing video call to C2
2. C2 answers the call. Receiving video frozen on both side. Video self preview active on both side. No audio on both side.
[wire_debug_report-c1.txt](https://github.com/wireapp/wire-android/files/678532/wire_debug_report-c1.txt)

**Comment by red-avtovo** [comment]
Audio call - outgoing call was not muted.
Outgoing call is not muted, when mute was enabled before answer
Bug doesn't reproduce on calling v2
C1@Nexus 5x, Android 7.0 (calling v3) [wire_audio_mute_report_c1.txt](https://github.com/wireapp/wire-android/files/678548/wire_audio_mute_report_c1.txt)
C2@Nexus 5, Android 6.0.1 (calling v3)
-- or --
C1@Nexus 5, Android 6.0.1 (calling v3)
C2@Nexus 5x, Android 7.0 (calling v3)

STR:
* C1 makes audio call
* C1 mutes his microphone
* C2 answers
* :collision: C2 hears all the sounds from C1

**Comment by deanrobertcook** [comment]
build again

**Comment by gregor** [comment]
build again

**Comment by gregor** [comment]
build again

**Comment by jspittka** [comment]
build again

**Comment by jspittka** [comment]
build again

**Comment by deanrobertcook** [comment]
build again

File: app/src/main/scala/com/waz/zclient/media/SoundController.scala [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+  def playCameraShutterSound() = {
+    setMediaPlaying(R.raw.camera)
+    setVibrating(R.array.camera)
+  }
+
+  /**
+    * @param play For looping patterns, this parameter will tell to stop vibrating if they have previously been started
+    */
+  private def setVibrating(patternId: Int, play: Boolean = true, loop: Boolean = false): Unit = {
```
Comment (hakonbo) [inline app/src/main/scala/com/waz/zclient/media/SoundController.scala]: I guess in theory you could end up with endless vibration if vibration setting is switched to off while a vibration request is currently looping. Pretty far fetched I guess, but would it hurt to do `vib.cancel()` in the catch all case?

**Comment by hakonbo** [comment]
build again

File: app/src/main/scala/com/waz/zclient/media/SoundController.scala (hunk 1 above)
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/media/SoundController.scala]: Hmm, the catch all case is the case in which the `Vibrator` isn't available to me, so there's nothing to cancel on...

**Comment by deanrobertcook** [comment]
build again

**Pull request merged**